core/src/testUtils/groovy/nu/yona/server/test/AppService.groovy

```diff
@@ -131,46 +159,59 @@ class AppService extends Service
 	def assertUserCreationResponseDetails(def response)
 	{
 		assertResponseStatusCreated(response)
-		assertUserWithPrivateData(response.responseData)
+		assertUserWithPrivateData(response.responseData, false)
 	}
 
 	def assertUserUpdateResponseDetails(def response)
 	{
 		assert response.status == 200
-		assertUserWithPrivateData(response.responseData)
+		assertUserWithPrivateData(response.responseData, false)
 	}
 
 	def assertUserGetResponseDetailsWithPrivateData(def response)
 	{
 		assertResponseStatusSuccess(response)
-		assertUserWithPrivateData(response.responseData)
+		assertUserWithPrivateData(response.responseData, false)
 	}
 
-	def assertUserGetResponseDetailsWithPrivateDataCreatedOnBuddyRequest(def response)
+	def assertUserGetResponseDetailsWithPrivateDataPinResetRequestedNotGenerated(def response)
 	{
 		assertResponseStatusSuccess(response)
 		assertUserWithPrivateData(response.responseData, true)
+		assert response.responseData.keySet() == PRIVATE_USER_PROPERTIES_NUM_CONFIRMED_BEFORE_ACTIVITY
+		assert response.responseData._links.keySet() == PRIVATE_USER_LINKS_NUM_CONFIRMED_PIN_RESET_REQUESTED_NOT_GENERATED
+		assert response.responseData._embedded.keySet() == PRIVATE_USER_EMBEDDED
 	}
 
-	def assertUserGetResponseDetailsWithoutPrivateData(def response)
+	def assertUserGetResponseDetailsWithPrivateDataPinResetRequestedAndGenerated(def response)
 	{
 		assertResponseStatusSuccess(response)
-		assertUserWithoutPrivateData(response.responseData)
+		assertUserWithPrivateData(response.responseData, true)
+		assert response.responseData.keySet() == PRIVATE_USER_PROPERTIES_NUM_CONFIRMED_BEFORE_ACTIVITY
+		assert response.responseData._links.keySet() == PRIVATE_USER_LINKS_NUM_CONFIRMED_PIN_RESET_REQUESTED_AND_GENERATED
+		assert response.responseData._embedded.keySet() == PRIVATE_USER_EMBEDDED
 	}
 
-	def assertUserWithPublicDataAndVpnProfile(user)
+	def assertUserGetResponseDetailsWithPrivateDataCreatedOnBuddyRequest(def response)
 	{
-		assertPublicUserData(user)
-		assertVpnProfile(user)
+		assertResponseStatusSuccess(response)
+		assertUserWithPrivateData(response.responseData, true, true)
+		assert response.responseData.keySet() == PRIVATE_USER_PROPERTIES_CREATED_ON_BUDDY_REQUEST
 	}
 
-	def assertUserWithPrivateData(user, boolean userCreatedOnBuddyRequest = false)
+	def assertUserGetResponseDetailsWithoutPrivateData(def response)
 	{
-		assertPublicUserData(user, userCreatedOnBuddyRequest)
-		assertPrivateUserData(user, userCreatedOnBuddyRequest)
+		assertResponseStatusSuccess(response)
+		assertPublicUserData(response.responseData, false, false)
 	}
 
-	def assertPublicUserData(def user, boolean userCreatedOnBuddyRequest)
+	def assertUserWithPrivateData(user, boolean skipPropertySetAssertion = true, boolean userCreatedOnBuddyRequest = false)
```

Comment:
Why do you by default skip the property set assertion? Because it fails current tests if set to true?

Reply:
No. The asserters are calling each other. The outer most one needs to do the property assertions, as they are the only ones with complete knowledge about what properties are expected.